File: Navtrack.Web/Model/Models/AssetModel.cs
using System.ComponentModel.DataAnnotations;

namespace Navtrack.Web.Model.Models
{
    public class AssetModel
    {
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }
        public int DeviceId { get; set; }
        public string DeviceType { get; set; }
    }
}

File: Navtrack.Web/ClientApp/src/services/Api/Model/AssetModel.ts
import { LocationModel, DefaultLocationModel } from "./LocationModel";

export type AssetModel = {
  id: number,
  name: string,
  deviceId: number,
  deviceType: string,
  location: LocationModel
};

export const DefaultAssetModel : AssetModel = {
  id: 0,
  name: "",
  deviceId: 0,
  deviceType: "",
  location: DefaultLocationModel
};